modules/katello_sync.py

```diff
@@ -107,115 +111,60 @@
-
-        product = self._entities.Product(self._server, name=name, organization=org)
-        response = product.search()
-
-        if len(response) == 1:
-            return response[0]
-        else:
-            self._module.fail_json(msg="No Product found for %s" % name)
-
-    def find_repository(self, name, product, organization):
-        product = self.find_product(product, organization)
+    from ansible.module_utils.ansible_nailgun_cement import (
+        create_server,
+        ping_server,
+        find_organization,
+        find_product,
+        find_repository,
+    )
 
-        repository = self._entities.Repository(self._server, name=name, product=product)
-        repository._fields['organization'] = entity_fields.OneToOneField(entities.Organization)
-        repository.organization = product.organization
-        response = repository.search()
+    from nailgun import entity_mixins
+except ImportError:
+    pass
 
-        if len(response) == 1:
-            return response[0]
-        else:
-            self._module.fail_json(msg="No Repository found for %s" % name)
+from ansible.module_utils.foreman_helper import ForemanAnsibleModule
 
-    def sync(self, product, organization, synchronous, repository=None):
-        if repository is None:
-            product = self.find_product(product, organization)
-            return product.sync(synchronous)
-        else:
-            repository = self.find_repository(repository, product, organization)
-            return repository.sync(synchronous)
+entity_mixins.TASK_TIMEOUT = 180000  # Publishes can sometimes take a long, long time
 
 
 def main():
-    module = AnsibleModule(
+    module = ForemanAnsibleModule(
         argument_spec=dict(
-            server_url=dict(required=True),
-            username=dict(required=True, no_log=True),
-            password=dict(required=True, no_log=True),
-            verify_ssl=dict(type='bool', default=True),
-            product=dict(required=True),
-            synchronous=dict(type='bool', default=True),
             organization=dict(required=True),
+            product=dict(required=True),
             repository=dict(),
+            synchronous=dict(type='bool', default=True),
         ),
         supports_check_mode=False,
     )
 
-    if has_import_error:
-        module.fail_json(msg=import_error_msg)
-
-    server_url = module.params['server_url']
-    verify_ssl = module.params['verify_ssl']
-    username = module.params['username']
-    password = module.params['password']
-    repository = module.params['repository']
-    product = module.params['product']
-    synchronous = module.params['synchronous']
-    organization = module.params['organization']
-
-    server = ServerConfig(
-        url=server_url,
-        auth=(username, password),
-        verify=verify_ssl
-    )
-    ng = NailGun(server, entities, module)
+    params = module.parse_params()
+
+    module.connect()
```

Comment:
do you need to re-record the tapes if you pass `ping=False` here? just curious :)

Reply:
No, still needs rerecording.